### PR TITLE
Treat no_changes_detected as a successful response

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- The user profile was being re-requested when no changes were detected, and so this PR primarily adds logic to make sure that the user endpoint is only re-requested when the user had a Bad Address Indicator present, and had updated their address to confirm that it was the correct address.
- Also updated the reducers for transactions to pull out the transactions from `transactionsAwaitingUpdate` when they complete one way or another. This wasn't being done before and resulted in some unexpected behavior.
- There were a couple areas that other forms use the contact info fields, and their tests had to be tweaked a bit for this change.
 
## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/63333

## Testing done

- Updated several tests for new logic.

- Locally, this could be tested by opening the edit screen for mobile phone, and using `1111111111` as the phone number. If you update the field with that value, the mock will respond with the No Changes Detected response, and you can view the network tab to see that the user endpoint is not re-requested.

## Screenshots

No ui changes, just logic / api call reduction

## What areas of the site does it impact?

Profile, some forms that rely on the profile shared components

## Acceptance criteria

- [x] user endpoint not re-requested when no changes detected and BAI is not present

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed (gonna work on cleaning those up next)

### CODE REVIEW

File: src/applications/appeals/995/tests/fixtures/mocks/telephone-update.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "data": {
     "attributes": {
-      "transaction_status": "RECEIVED",
-      "transaction_id": "c049d895-ecdf-40a4-ac0f-7947a06ea0c2",
+      "transactionStatus": "RECEIVED",
+      "transactionId": "c049d895-ecdf-40a4-ac0f-7947a06ea0c2",
```
Comment (adamwhitlock1) [inline src/applications/appeals/995/tests/fixtures/mocks/telephone-update.json:5]: mock responses should be camelCased for the reducers for profile to work correctly.

File: src/applications/appeals/996/tests/hlr-contact-loop.cypress.spec.js [hunk 2]
```diff
@@ -135,9 +135,9 @@ describe('HLR contact info loop', () => {
       `${BASE_URL}/edit-contact-information-mobile-phone`,
     );
 
-    cy.findByLabelText(/mobile phone/i)
-      .clear()
-      .type('8885551212');
```
Comment (adamwhitlock1) [inline src/applications/appeals/996/tests/hlr-contact-loop.cypress.spec.js:140]: this is a cypress error now, saying its unsafe to do multiple chaining of commands

File: src/applications/personalization/profile/components/ProfileInformationEditView.jsx [hunk 3]
```diff
@@ -95,13 +94,6 @@ export class ProfileInformationEditView extends Component {
     ) {
       window.clearInterval(this.interval);
     }
-    // if a transaction was created that was immediately successful (for example
-    // when the transaction's status is `COMPLETED_NO_CHANGES_DETECTED`),
-    // immediately exit edit view and clear the transaction request so it can be triggered again
-    if (isSuccessfulTransaction(this.props.transaction)) {
-      this.props.openModal(null);
-      this.props.clearTransactionRequest(this.props.fieldName);
-    }
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/ProfileInformationEditView.jsx:104]: Removed bc now the transaction is treated as successful if no changes are detected

File: src/platform/user/profile/vap-svc/actions/transactions.js [hunk 4]
```diff
@@ -159,16 +160,21 @@ export function refreshTransaction(
...
     transaction?.data?.attributes?.transactionStatus ===
     TRANSACTION_STATUS.COMPLETED_NO_CHANGES_DETECTED;
 
   if (noChangesDetected) {
-    const forceCacheClear = true;
-    await dispatch(refreshProfile(forceCacheClear));
+    if (hasBadAddress(state) && fieldName === FIELD_NAMES.MAILING_ADDRESS) {
+      const forceCacheClear = true;
+      await dispatch(refreshProfile(forceCacheClear));
+    }
```
Comment (adamwhitlock1) [inline src/platform/user/profile/vap-svc/actions/transactions.js:177]: main logic for figuring out if the user endpoint needs to be refreshed

**Review by allisonlu — APPROVED** [review]
:shipit:

**Pull request merged**